src/k8s-engine/lima.ts

```diff
@@ -330,6 +331,48 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
     return Promise.resolve(null);
   }
 
+  /**
+   * Check if the base (alpine) disk image is out of date; if yes, update it
+   * without removing existing data.  This is only ever called from updateConfig
+   * to ensure that the passed-in lima configuration is the one before we
+   * overwrote it.
+   */
+  protected async updateBaseDisk(currentConfig: LimaConfiguration) {
+    // Lima does not have natively have any support for this; we'll need to
+    // reach into the configuration and:
+    // 1) Figure out what the old base disk version is.
+    // 2) Confirm that it's out of date.
+    // 3) Change out the base disk as necessary.
+    // Unfortunately, we don't have a version string anywhere _in_ the image, so
+    // we will have to rely on the path in lima.yml instead.
+
+    const images = currentConfig.images.map(i => path.basename(i.location));
+    // We had a typo in the name of the image; it was "alpline" instead of "alpine".
+    const versionMatch = images.map(i => /^alpl?ine-lima-v([0-9.]+)-rd-/.exec(i)).find(defined);
+
+    if (!versionMatch) {
+      console.log(`Could not find base image version from ${ images }; skipping update of base images.`);
+
+      return;
+    }
+    const existingVersion = semver.coerce(versionMatch[1]);
+
+    if (!existingVersion || (semver.coerce(IMAGE_VERSION)?.compare(existingVersion) ?? 0) <= 0) {
+      return;
+    }
+    console.log(`Attempting to update base image from ${ existingVersion } to ${ IMAGE_VERSION }...`);
+
+    const diskPath = path.join(paths.lima, MACHINE_NAME, 'basedisk');
+
+    await fs.promises.copyFile(this.baseDiskImage, diskPath);
+    // The config file will be updated in updateConfig() instead; no need to do it here.
```

Comment:
I assume this whole update logic will only ever be entered while the instance is stopped, right?

Reply:
Updated to force stop the VM in a1af7ac5d957f3fafea79662a3e88ad10d2140a3